topdown/cache/cache.go

```diff
@@ -301,3 +326,81 @@ func (c *cache) cleanStaleValues() (dropped int) {
 	}
 	return dropped
 }
+
+type ValueCache interface {
+	Get(key string) (value any, found bool)
```

Comment:
One additional thought: the `InterQueryCache` uses `key ast.Value` as key. Should this new cache do the same? `string` type keys aren't necessarily the most convenient type for custom built-ins that might want to use this.

Reply:
Good point. Updated keys to be `ast.Value` type.